arangod/Transaction/Manager.cpp

```diff
@@ -120,9 +121,12 @@ void Manager::unregisterFailedTransactions(std::unordered_set<TRI_voc_tid_t> con
 
 void Manager::registerTransaction(TRI_voc_tid_t transactionId,
                                   std::unique_ptr<TransactionData> data,
-                                  bool isReadOnlyTransaction) {
-  if (!isReadOnlyTransaction && !isFollowerTransactionId(transactionId)) {
+                                  bool isReadOnlyTransaction,
+                                  bool isFollowerTransaction) {
+  if (!isReadOnlyTransaction && !isFollowerTransaction) {
```

Comment:
I think we should an an assertion here, so that if `isFollowerTransaction` is true, the method `isFollowerTransactionId(transactionId)` should also return true or `isLegacyTransactionId(transactionId)` should return true.

Reply:
Added.